### DISCUSSION TITLE
Build and run FIO in QNX

### DISCUSSION THREAD

**Discussion by arthur-simchaev-sndk** [axboe/fio#1813] [Q&A]
Hi All. 

I built fio for QNX according to 946733c6  commit "Support QNX OS platform".
Unfortinualy upon running the tool in NUC which runs QNX 7.1 OS I get Illegal instruction error:
_**Process 512020 (fio) terminated SIGILL code=1 fltno=1 ip=0000000000400577 ref=0000000000000000
Illegal instruction (core dumped)_**_
 Does anybody faced with this problem and know the root cause?

Regards
Arthur

**Comment by axboe** [comment]
It's likely because you're using a different CPU that what the original author did. What platform are you on?

**Comment by axboe** [comment]
It would be helpful if you run:

```
$ gdb ./fio <core file being dumped>
```

and include the output of the backtrace, as that will tell us where it actually its SIGILL. Right now, with what you provided, all that is known is that it's an illegal instruction, but no idea where it occured.

**Comment by arthur-simchaev-sndk** [comment]
I built it on Linux and run it on NUC (x86_64).
The target OS configurate to QNX.
The tool configuration is follows:
./configure --build-static --disable-shm --prefix=2_x86_qnx_stattic_install_dir/ --cc=x86_64-pc-nto-qnx7.1.0-gcc

The generated file :
file fio
fio: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[md5/uuid]=d79239e0e2cc896f7174c1971956c335, with debug_info, not stripped

Unfortinualy I cannot run gdb in QNX OS in order to get the backtrace

**Reply by axboe** [reply]
Can't help you without a backtrace.

**Reply by arthur-simchaev-sndk** [reply]
Thank you, will ask QNX

**Reply by axboe** [reply]
I'll close this for now, re-open when you have more info.
